Gems/Atom/RHI/Code/Source/RHI/RHIUtils.cpp

```diff
@@ -162,6 +162,13 @@ namespace AZ::RHI
         return graphicsDevMode;
     }
 
+    const AZ::Name& GetDefaultSupervariantNameWithNoFloat16Fallback()
+    {
+        const static AZ::Name DefaultSupervariantName{ "" };
```

Comment:
Do you not need to handle Masa/NoMsaa here as well? You can also try turning masa off to ensure it doesn't crash.

Reply:
I think you can use GetSupervariantName to check if the app is using NoMSAA super variant or not.

Reply:
I believe this is already handled automatically; if msaa is enabled, in `RPISystem::SetApplicationMultisampleState` the function `ShaderSystemInterface::Get()->SetSupervariantName(...)` is called, and later in `ShaderAsset::GetSupervariantIndex(...)` ([code](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Reflect/Shader/ShaderAsset.cpp#L163)) the "NoMSAA" from `shaderSystemInterface->GetSupervariantName()` is appended to the user-provided supervariant name.
Running the editor with and without msaa (game.cfg `r_multiSampleCount=1` or `4`) both work, although the reflections and the shapes from auxgeometry are always drawn without msaa.